packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,209 @@
+/// result.
+///
+/// This is a core framework widget with very basic UI. Try using [Autocomplete]
+/// or [AutocompleteCupertino] before resorting to this widget.
+class AutocompleteCore<T> extends StatefulWidget {
+  /// Create an instance of AutocompleteCore.
+  ///
+  /// [autocompleteController], [buildField], and [buildResults] must not be
+  /// null.
+  const AutocompleteCore({
+    @required this.autocompleteController,
+    @required this.buildField,
+    @required this.buildResults,
+  }) : assert(autocompleteController != null),
+       assert(buildField != null),
+       assert(buildResults != null);
+
+  /// The controller that provides access to the main autocomplete state and
+  /// logic.
+  final AutocompleteController<T> autocompleteController;
+
+  /// Builds the field that is used to input the query.
+  final AutocompleteFieldBuilder buildField;
+
+  /// Builds the selectable results of searching.
+  final AutocompleteResultsBuilder<T> buildResults;
+
+  @override
+  _AutocompleteCoreState<T> createState() =>
+      _AutocompleteCoreState<T>();
+}
+
+class _AutocompleteCoreState<T> extends State<AutocompleteCore<T>> {
+  T _selection;
+
+  void _onChangeResults() {
+    setState(() {});
+  }
+
+  void _onChangeQuery() {
+    if (widget.autocompleteController.textEditingController.value.text == _selection) {
+      return;
+    }
+    setState(() {
+      _selection = null;
+    });
+  }
+
+  void onSelected (T result) {
+    setState(() {
+      _selection = result;
+      widget.autocompleteController.textEditingController.text = result.toString();
+    });
+  }
+
+  void _listenToController(AutocompleteController<T> autocompleteController) {
+    autocompleteController.results.addListener(_onChangeResults);
+    autocompleteController.textEditingController.addListener(_onChangeQuery);
+  }
+
+  void _unlistenToController(AutocompleteController<T> autocompleteController) {
+    autocompleteController.results.removeListener(_onChangeResults);
+    autocompleteController.textEditingController.removeListener(_onChangeQuery);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _listenToController(widget.autocompleteController);
+  }
+
+  @override
+  void didUpdateWidget(AutocompleteCore<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.autocompleteController != oldWidget.autocompleteController) {
+      _unlistenToController(oldWidget.autocompleteController);
+      _listenToController(widget.autocompleteController);
+    }
+  }
+
+  @override
+  void dispose() {
+    _unlistenToController(widget.autocompleteController);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        widget.buildField(
+          context,
+          widget.autocompleteController.textEditingController,
+        ),
+        if (_selection == null)
+          // TODO(justinmc): should this expanded be here?
```

Comment:
I feel like in most cases, this does make sense. I'm trying to imagine an odd layout where a developer would not want the results to be expanded, but can't off the top of my head.